lib/services/role.go

```diff
@@ -154,13 +154,14 @@ func RoleForUser(u types.User) types.Role {
 			BPF:               defaults.EnhancedEvents(),
 		},
 		Allow: types.RoleConditions{
-			Namespaces:            []string{defaults.Namespace},
-			NodeLabels:            types.Labels{types.Wildcard: []string{types.Wildcard}},
-			AppLabels:             types.Labels{types.Wildcard: []string{types.Wildcard}},
-			GroupLabels:           types.Labels{types.Wildcard: []string{types.Wildcard}},
-			KubernetesLabels:      types.Labels{types.Wildcard: []string{types.Wildcard}},
-			DatabaseServiceLabels: types.Labels{types.Wildcard: []string{types.Wildcard}},
-			DatabaseLabels:        types.Labels{types.Wildcard: []string{types.Wildcard}},
+			Namespaces:                   []string{defaults.Namespace},
+			NodeLabels:                   types.Labels{types.Wildcard: []string{types.Wildcard}},
+			AppLabels:                    types.Labels{types.Wildcard: []string{types.Wildcard}},
+			GroupLabels:                  types.Labels{types.Wildcard: []string{types.Wildcard}},
+			KubernetesLabels:             types.Labels{types.Wildcard: []string{types.Wildcard}},
+			DatabaseServiceLabels:        types.Labels{types.Wildcard: []string{types.Wildcard}},
+			DatabaseLabels:               types.Labels{types.Wildcard: []string{types.Wildcard}},
+			SAMLIdPServiceProviderLabels: types.Labels{types.Wildcard: []string{types.Wildcard}},
```

Comment:
As I was reviewing this, I kept thinking, SAML apps are basically just apps, so can we just continue using existing `app_labels` rather than introducing yet another type of labels? I left a similar comment on the RFD as well: https://github.com/gravitational/teleport.e/pull/3476#pullrequestreview-1929915151.

Reply:
I did consider that, and we already show SAML apps as "apps" in the UI. But its not straightforward enough as we treat SAML service provider resource as an independent resource in the majority of codebase and in the storage backend. It has its own RBAC verbs and configuration flow. We are talking about a not so simple overhaul.

And the main issue I believe we will have it with that is we will still end up with a confusing UX for admins as we have a different workflow to manage `saml_idp_service_provider` resource but a different workflow to manage RBAC.

So this PR is based on plan to just change the "view layers" to show SAML apps and apps and let admin workflow based on `saml_idp_service_provider`. This is what we do today and this label matcher was added in the same context.

That said, I would still want to hear your opinion on what is better for the longer run.